packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1544,12 +1550,23 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
 
   @override
   void handleEvent(PointerEvent event, BoxHitTestEntry entry) {
-    if (ignorePointer)
-      return;
     assert(debugHandleEvent(event, entry));
-    if (event is PointerDownEvent && onSelectionChanged != null) {
-      _tap.addPointer(event);
-      _longPress.addPointer(event);
+    if (event is PointerDownEvent) {
+      // Checks if there is any gesture recognizer in the text span.
+      final Offset offset = entry.localPosition;
+      _layoutText(minWidth: constraints.minWidth, maxWidth: constraints.maxWidth);
```

Comment:
since this is the third time we have this exact line with the exact same logic, can we factor this out into a private method that uses this exact set of arguments? we need to make sure that if we ever change layout, paint, or hit test, we change all three